.ci/common.sh

```diff
@@ -51,20 +51,8 @@ check_core_dumps() {
 }
 
 setup_deps() {
-	sudo pip install --upgrade pip
-	sudo pip install neovim
-
-	# For pip3
-	# https://github.com/travis-ci/travis-ci/issues/1528
-	# sudo apt-get install -q python3.3-dev
-	# curl -Ss http://python-distribute.org/distribute_setup.py | sudo python3
-	# curl -Ss https://raw.github.com/pypa/pip/master/contrib/get-pip.py | sudo python3
-	# sudo pip3.3 install neovim
-
 	if [ "$BUILD_NVIM_DEPS" != "true" ]; then
```

Comment:
And maybe put in an else:

```bash
...
elif [ "$TRAVIS_OS_NAME" = "linux" ]; then
  # Make sure deps are up-to-date before proceeding
  # Remove sources from cache to force download on dependency changes
  rm -rf .deps/build/src
  make deps
fi
```

Otherwise, if a dependency is updated to a new version, CMake will register that something has changed and rebuild, but skip fetching the new sources (due to [this](https://github.com/neovim/neovim/blob/master/third-party/cmake/DownloadAndExtractFile.cmake#L28)) and just build the old version again. Maybe the `rm` should be put in [`before_cache`](https://github.com/neovim/neovim/pull/2340#issuecomment-91253389) instead..

Maybe there's a more clever approach.. what do you think @jszakmeister?